server/service/openapi/openapi.go

```diff
@@ -0,0 +1,288 @@
+package openapi
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/invopop/jsonschema"
+)
+
+type endpoint struct {
+	name string
+	verb string
+	path string
+	req  interface{}
+	res  interface{}
+}
+
+type Document struct {
```

Comment:
Ignoring the internal `endpoints` field and the `Render` method which were just added as a quick way to go from our handlers to an openapi document generation, this file contains the data structures that can marshal openapi v3 documents. Only a subset of the complete spec is supported, but that's arguably the most important parts such as the routes, parameters, body, and responses (unimplemented fields from the spec are added but commented-out).